instance/cloudServerCreate.go

```diff
@@ -204,20 +223,15 @@ func (ci *Client) CloudServerCreate(params *CloudServerCreateParams) (string, er
 		createArgs["features"].(map[string]interface{})["WinAV"] = params.WinAv
 		createArgs["features"].(map[string]interface{})["WindowsLicense"] = "Windows"
 		if params.Type == "SS.VPS" {
-			createArgs["type"] = "SS.VPS.WIN"
+			params.Type = "SS.VPS.WIN"
 		}
 		if params.MsSql == "" {
 			params.MsSql = "None"
 		}
 		var coreCnt int
 		if params.Vcpu == -1 {
 			// standard config_id create, fetch configs core count and use it
-			var details apiTypes.CloudConfigDetails
-			if err := ci.CallLwApiInto("bleed/storm/config/details",
-				map[string]interface{}{"id": params.ConfigId}, &details); err != nil {
-				return "", err
-			}
-			coreCnt = cast.ToInt(details.Vcpu)
+			coreCnt = cast.ToInt(configDetails.Vcpu)
```

Comment:
is it possible for configDetails to not be defined at this spot?  it looks like `isWindows` and `params.ConfigId != -1` are mutually exclusive

Reply:
if `ConfigId` is passed (so not -1) Vcpu should never have a passed value (of not -1)

Reply:
oic, that means validation should catch that well before

Reply:
yeah

Reply:
i'll adjust the validation, looks like it doesnt complain if you pass `config-id` flag but you do private parent flags like vcpu

Reply:
i updated the validation here d4c933132395822f9d328fbcd3bd5613abe542b4